### PR TITLE
#71 - add newsletter views

### LINKED ISSUE #71: Newsletter views
The aim of this task is to make simple views to the newsletter

### PR DESCRIPTION
**Pull request by JakubZakowicz**
I have created simple views for newsletter.

- Here is a page where user passes his email to subscribe or unsubscribe newsletter.
screenshot-localhost-2022 05 05-00_45_12

- Here user can choose his preferences.
screenshot-localhost-2022 04 25-20_38_46

- And here is message page.
newsletter_message

This should close #71

### CODE REVIEW

File: resources/views/newsletter/subscribe.blade.php [hunk 1]
```diff
@@ -1,25 +1,62 @@
...
+                <div class="divide-y divide-gray-200">
+                  <div class="relative flex items-start py-4">
+                    <div class="min-w-0 flex-1 text-sm">
+                      <label for="meetups" class="font-medium text-white">New Meetups</label>
+                      <p id="mettups-description" class="text-indigo-200">Get notified about new meetups that might
+                        interest you</p>
+                    </div>
+                    <div class="ml-3 flex items-center h-5">
+                      <input id="meetups" name="type[]" checked type="checkbox" value={{ AvailableNewsletter::Meetups->value }}
+                        class="h-[18px] w-[18px] text-indigo-600 border-gray-300 rounded default:border-gray-300 checked:border-gray-300 checked:border-2">
```
Comment (EwelinaLasowy) [inline resources/views/newsletter/subscribe.blade.php:37]: Is really a good idea to write like this? `h-[18px] w-[18px]`

**Review by EwelinaLasowy — COMMENTED** [review]
We have here a strong indigo color as the background - as a result, indigo buttons are not very visible, although a bit brighter. Maybe buttons should be white with indigo text? Also think of check boxes.

File: resources/views/newsletter/subscribe.blade.php (hunk 1 above)
Comment (JakubZakowicz) [reply resources/views/newsletter/subscribe.blade.php:37]: Default tailwind widths and heights are not very accurate, so i picked this method. I just read about this, and it's nothing wrong with using arbitrary values, when styling is not repeatable.

Comment (JakubZakowicz) [reply resources/views/newsletter/subscribe.blade.php]: I changed the buttons. I think they are more visible now. As to checkboxes. They are very difficult to style. How they look now is the best result I managed to accomplish with @Majkulnn. Alternatively we can fix them later.

As for subscription. We decided with you that newsletter should be available for logged and and unlogged users. That's why there are subscribe and unsubscribe buttons.

File: resources/views/emails/newsletters/news.blade.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+<p>New news has been created</p>
+<p>Title {{$title}}</p>
+@component('mail::button', ['url' => $url])
+    Click here for details
+@endcomponent
+<p>if you want to unsubscribe click here</p>
+@component('mail::button', ['url' => route("newsletter.unsubscribe",["email" => $email])])
+    Click here to unsubscribe
+@endcomponent
+@endcomponent
```
Comment (kamilpiech97) [inline resources/views/emails/newsletters/news.blade.php:12]:
```suggestion
@endcomponent

```

File: src/Http/Controllers/NewsletterSubscriberController.php [hunk 3]
```diff
@@ -47,6 +47,13 @@ public function update(NewsletterUpdateRequest $request): View
         return view("newsletter.dashboard")->with("message", "You are now subscribed.");
     }
 
+    public function delete(): View
+    {
+        $email = $_GET["email"];
+
+        return view("newsletter.unsubscribe")->with("email", $email);
```
Comment (kamilpiech97) [inline src/Http/Controllers/NewsletterSubscriberController.php:54]: U can use `request` object and get `email` from him instead of using `$_GET`.

**Review by kamilpiech97 — CHANGES_REQUESTED** [review]

**Review by kamilpiech97 — CHANGES_REQUESTED** [review]

**Comment by krzysztofrewak** [comment]
Please resolve conflicts before further code review.

**Review by EwelinaLasowy — APPROVED** [review]

**Review by krzysztofrewak — APPROVED** [review]

**Review by kamilpiech97 — APPROVED** [review]

**Pull request merged**